src/test/java/org/takes/tk/TkProxyTest.java

```diff
@@ -30,26 +30,140 @@
+                @Override
+                public void exec(final URI home) throws IOException {
+                    MatcherAssert.assertThat(
+                        TkProxyTest.this.actResponse(home, "GET", headers),
+                                    Matchers.startsWith(OK)
+                    );
+                }
+            }
+        );
+        new FtRemote(new TkFixed(HELLO_WORLD)).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    MatcherAssert.assertThat(
+                        TkProxyTest.this.actResponse(home, "POST", headers),
+                        Matchers.startsWith(OK)
+                    );
+                }
+            }
+        );
+        new FtRemote(new TkFixed(HELLO_WORLD)).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    MatcherAssert.assertThat(
+                        TkProxyTest.this.actResponse(home, "OPTIONS", headers),
+                        Matchers.startsWith(OK)
+                    );
+                }
+            }
+        );
+        new FtRemote(new TkFixed(HELLO_WORLD)).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    MatcherAssert.assertThat(
+                        TkProxyTest.this.actResponse(home, "PUT", headers),
+                        Matchers.startsWith(OK)
+                    );
+                }
+            }
+        );
+        new FtRemote(new TkFixed(HELLO_WORLD)).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    MatcherAssert.assertThat(
+                        TkProxyTest.this.actResponse(home, "DELETE", headers),
+                        Matchers.startsWith(OK)
+                    );
+                }
+            }
+        );
+        new FtRemote(new TkFixed(HELLO_WORLD)).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    try {
+                        TkProxyTest.this.actResponse(home, "INVALID", headers);
+                        MatcherAssert.assertThat("IOException expected", false);
+                    } catch (final IOException exception) {
+                        MatcherAssert.assertThat("IOException - OK", true);
+                    }
+                }
+            }
+        );
+    }
+
+    /**
+     * TkProxy can proxy original response headers.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void actsOnResponseHeaders() throws Exception {
+        new FtRemote(
+                new TkFixed(
+                        HELLO_WORLD
```

Comment:
@aschworer In this method, you will need string value twice - use local variable.
The main goal - each test should be self-contained. Nothing should be shared between tests unless it's really needed.